discord/ext/commands/core.py

```diff
@@ -1305,13 +1310,15 @@ def has_role(item):
 
     def predicate(ctx):
         if not isinstance(ctx.channel, discord.abc.GuildChannel):
-            return False
+            raise NoPrivateMessage('This command cannot be used in private messages.')
```

Comment:
If you're going to reuse this everywhere then it's probably for the best that this message is stored inside `NoPrivateMessage.__init__`.